File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        static int steps[46]{};
        steps[1] = 1;
        steps[0] = 1;
        for (int i = 2; i <= n; ++i)
            steps[i] = steps[i - 1] + steps[i - 2];
        return steps[n];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut dp = vec![];
        dp.push(1);
        dp.push(1);
        for i in 2..=n {
            dp.push(dp[i as usize - 1] + dp[i as usize - 2]);
        }
        
        dp[n as usize]
    }
}